public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.tsx

```diff
@@ -0,0 +1,144 @@
+import { SelectableValue } from '@grafana/data';
+import { Alert, InlineField, Input, Select } from '@grafana/ui';
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import { AzureMonitorQuery, AzureQueryType } from '../../types';
+import LogsQueryEditor from '../LogsQueryEditor';
+import DataSource from '../../datasource';
+import useLastError from '../../utils/useLastError';
+import { Space } from '../Space';
+import { migrateStringQueriesToObjectQueries } from '../../grafanaTemplateVariableFns';
+
+const AZURE_QUERY_VARIABLE_TYPE_OPTIONS = [
+  { label: 'Grafana Query Function', value: AzureQueryType.GrafanaTemplateVariableFn },
+  { label: 'Logs', value: AzureQueryType.LogAnalytics },
+];
+
+const GrafanaTemplateVariableFnInput = ({
+  query,
+  updateQuery,
+  datasource,
+}: {
+  query: AzureMonitorQuery;
+  updateQuery: (val: AzureMonitorQuery) => void;
+  datasource: DataSource;
+}) => {
+  const [inputVal, setInputVal] = useState('');
+  useEffect(() => {
+    setInputVal(query.grafanaTemplateVariableFn?.rawQuery || '');
+  }, [query.grafanaTemplateVariableFn?.rawQuery]);
+  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setInputVal(event.target.value);
+  };
+  const onBlur = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
Instead of onBlur, i think this should be debounced/throttled in `onChange`.

Reply:
I'll leave it up to you to decide on what's best. I don't think onBlur is the _worst_,  _personally_ I would prefer onChange, but you can have a different opinion that you go with :)

CodeEditor uses onBlur because it lacks an onChange event, fwiw.

I agree - standardising on this seems like it would be helpful!